src/appleseed.studio/mainwindow/rendering/renderwidget.cpp

```diff
@@ -422,5 +428,29 @@ void RenderWidget::paintEvent(QPaintEvent* event)
     m_painter.end();
 }
 
+void RenderWidget::dragEnterEvent(QDragEnterEvent* event)
+{
+    if (event->mimeData()->hasFormat("text/plain") && m_project.has_trace_context())
```

Comment:
It looks like you only need the project here. I wonder if it wouldn't be simpler to check if the trace context exists elsewhere, that way the only responsability of `RenderWidget` would be to accept drag & drop events and emit the right signal. It sounds simpler to me.

Reply:
The reason why I check if the trace exist here, it is because it change the way the cursor is visible.
If no trace exist, the icon will stay as "action non available" and the drag and drop will no work.
If you prefer that we let the user drop the item even if it is not available, I will change it

Reply:
I think it would be cleaner, also it would allow to print a log message explaining why drag and drop is not available (versus just changing the icon right now). What do you think?

Reply:
I like the idea to have a visual feedback which say that it is not a legal action but it is true that it doesnt explain why it is illegal. I dont really have a opinion on this, it is why both check have beeb done

Reply:
The move _is_ legal, it's just not possible for a technical reason that might disappear one day. It's also not clear from a user's point of view why it's sometimes legal and sometimes illegal. I'm in favor of removing this check, simplifying the code accordingly and printing a clear error message when the move ends up being impossible.